R/spark_shell.R

```diff
@@ -7,6 +7,68 @@ shell_connection_validate_config <- function(config) {
   config
 }
 
+databricks_connection <- function(config, extensions) {
+  tryCatch({
+    gatewayPort <- as.numeric(SparkR:::callJStatic("com.databricks.backend.daemon.driver.RDriverLocal",
```

Comment:
Actually, this comes before the connection is opened so `invoke_static` won't work here. This call assumes that `SparkR` is configured in their cluster and uses `SparkR` to make the `RDriverLocal` call.

To have this be a clean call, we could either:

1. Always start the `sparklyr` backend in databricks and the port is read from a config.
2. Use some other mechanism to start `sparklyr` in databricks, say, with a call to a REST endpoint.

In the sort term, we would have to mask the call to `SparkR` with something like:

```
if (.rs.isPackageInstalled("SparkR")) {
      callSparkR <- get("callJStatic", envir = asNamespace("SparkR"))
      gatewayPort <- as.numeric(callSparkR(...))
}
```

otherwise, we won't be able to push this to CRAN since `SparkR` is not a known package.